src/components/UploadDropzone.tsx

```diff
@@ -0,0 +1,198 @@
+      fileRejections.map(({ file, errors }: FileRejections) =>
+        setRejectedFileItems((prev) => [...prev, { file, errors }]),
+      )
+    }
+  }, [acceptedFiles, fileRejections])
+
+  useEffect(() => {
+    setData(acceptedFileItems)
+  }, [acceptedFileItems, setData])
+
+  function formatFileSize(size: number) {
+    const formattedSize: number = size / 1000
+    if (formattedSize > 1000) return `${(formattedSize / 1000).toFixed(2)} MB`
+
+    return `${formattedSize.toFixed(2)} KB`
+  }
+
+  function getErrorMessage() {
+    if (maxFilesToUpload && fileRejections.length > maxFilesToUpload)
+      return `Too many files selected (maximun files: ${maxFilesToUpload})`
+    if (isOnlyImagesAllowed) return 'upload jpg, png, gif or svg image'
+    return 'upload PDF documents'
+  }
+
+  return (
+    <div>
+      <Container
+        {...getRootProps({ isDragActive, isDragAccept, isDragReject, className: 'dropzone disabled' })}
+      >
+        <input {...getInputProps()} />
+        <p>{`Drag 'n drop some${
+          isOnlyImagesAllowed && ' image'
+        } files here, or click to select files`}</p>
+      </Container>
+      {(acceptedFileItems.length > 0 || rejectedFileItems.length > 0) && (
+        <section className="section">
+          <aside>
+            {acceptedFileItems.length > 0 && (
+              <Section>
+                <h4 style={{ color: 'green' }}>Accepted files</h4>
+                {acceptedFileItems.map((file: File) => (
+                  <ListItem key={file.name}>
+                    {isOnlyImagesAllowed && (
+                      <img
+                        src={URL.createObjectURL(file)}
+                        width={40}
+                        height={40}
+                        alt={file.name}
+                        style={{ marginRight: '10px' }}
+                      />
+                    )}
+                    <span style={{ fontSize: 12 }}>
+                      {file.name} - {formatFileSize(file.size)}
+                    </span>
+                  </ListItem>
+                ))}
+              </Section>
+            )}
+            {rejectedFileItems.length > 0 && (
+              <Section>
+                <h4 style={{ color: 'red' }}>{`Rejected files - ${getErrorMessage()}`}</h4>
+                {rejectedFileItems.map(({ file, errors }: FileRejections) => (
+                  <ListItem key={file.name}>
+                    <span style={{ fontSize: 12 }}>{file.name}</span>
+                  </ListItem>
+                ))}
+              </Section>
+            )}
+          </aside>
+        </section>
+      )}
+      <ButtonsWrapper>
+        <Button label="cancel" onClick={onCancel} override={{ marginRight: 16 }} />
+        <Button
+          label="save"
+          type={ButtonType.Primary}
+          onClick={() => {
+            onSubmit()
+            onCancel()
+          }}
+        />
+      </ButtonsWrapper>
+    </div>
+  )
+}
+
+const getColor = (props: any) => {
```

Comment:
Next step would be to make this styleable via the theme. But for a first version that's okay in my opinion